main.cpp

```diff
@@ -26,7 +26,6 @@ int main(int argc, char **argv)
 
         // Ensure only one filter
         if (getopt(argc, argv, filtersOptions) != -1) {
```

Comment:
``` getopt() ``` shall return the next option character specified on the command line.

The first  **getopt()** reads the 1st and single filter flag specified at the command line and the flag i.e either of ``` b / e / g / r / s ``` will be returned and stored on the **filter** variable which will be later used for validation.

The second **getopt()** should return ```-1``` as the only one filter option has already been read and no more option should be there. If the user provides multiple filter options then it would read that option and return the corresponding integer of that char rather than -1 and the select statement would be true and an exception would be thrown.